src/ctap/response.rs

```diff
@@ -385,17 +384,17 @@ mod test {
             ResponseData::AuthenticatorGetAssertion(get_assertion_response).into();
         let expected_cbor = cbor_map_options! {
             0x01 => cbor_map! {
-                "type" => "public-key",
                 "id" => vec![0x2D, 0x2D, 0x2D, 0x2D],
+                "type" => "public-key",
```

Comment:
I don't understand why the macro input is sometimes reordered. Because it's not always sorted so it doesn't seem to be that.

Reply:
It should always be sorted. If there is something missing, please let me know. Maybe it is just the non-intuitive way CBOR works. Its sorting rules are:

> If the major types are different, the one with the lower value in numerical order sorts earlier.
> If two keys have different lengths, the shorter one sorts earlier;
> If two keys have the same length, the one with the lower value in (byte-wise) lexical order sorts earlier.

Reply:
Oh I see. Yes I didn't know that CBOR sorts strings differently than lexicographical order (which is the common sorting). Sounds good.